src/pages/CourseSelectionActivity.tsx

```diff
@@ -98,25 +152,33 @@ const CourseSelectionActivity: ActivityComponentType<CourseSelectionActivityPara
                 <div className="mt-10 flex w-full flex-1 flex-col gap-3 overflow-auto px-12">
                   {params.type === 'majorElective' && (
                     <div className="flex gap-1.5">
-                      {['1학년', '2학년', '3학년', '4,5학년'].map((grade) => (
+                      {gradeSelection.map((grades) => (
                         <GradeChip
-                          onClickGradeChip={onClickGradeChip(grade)}
-                          key={grade}
-                          isSelected={grade.includes(selectedGrade)}
-                          grade={grade}
+                          onClickGradeChip={onClickGradeChip(grades)}
+                          key={grades.join(', ')}
+                          isSelected={grades.join(',') === selectedGrades.join(',')}
+                          grades={grades}
                         />
                       ))}
                     </div>
                   )}
-                  <div className="flex flex-1 flex-col gap-3.5 overflow-auto">
-                    {courses.map((course) => (
-                      <CourseListItem
-                        onClickCourseItem={onClickCourseItem}
-                        isSelected={selectedCourseIds.includes(course.courseId)}
-                        key={course.courseId}
-                        course={course}
-                      />
-                    ))}
+                  <div className="overflow-auto">
+                    <div className="flex flex-1 flex-col gap-3.5">
+                      {courses.map((course) => (
+                        <CourseListItem
+                          onClickCourseItem={onClickCourseItem}
+                          isSelected={
+                            !!selectedCourses.find(
+                              ({ courseName, professorName }) =>
+                                course.courseName === courseName &&
+                                course.professorName === professorName,
+                            )
+                          }
```

Comment:
```suggestion
                          isSelected={selectedCourses.some((c) => isSameCourse(c, course))}
```
위에 사용한 `isSameCourse()` 함수를 사용하면 이렇게 줄일 수 있을 것 같아요